File: 백준/Silver/18258. 큐 2/큐 2.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.LinkedList;
import java.util.Queue;
import java.util.StringTokenizer;

public class Main {

	public static void main(String[] args) throws NumberFormatException, IOException {
		Queue<Integer> queue = new LinkedList<>();
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

		
		int N = Integer.parseInt(br.readLine());
		
		// 마지막 값은 queue로 불러올 수 없으니 저장
		int last = 0;
		
		for(int i=0;i<N;i++) {
			StringTokenizer st = new StringTokenizer(br.readLine());
			String str = st.nextToken();
			
			if(str.equals("push")) {
				int a = Integer.parseInt(st.nextToken());
				queue.offer(a);
				last = a;
				
			}
			else if(str.equals("pop")) {
				if(!queue.isEmpty()) {
					bw.write(queue.poll() + "\n");
				}
				else {
					bw.write(-1 + "\n");
				}
			}
			else if(str.equals("size")) {
				bw.write(queue.size() + "\n");
			}
			else if(str.equals("empty")) {
				if(queue.isEmpty()) {
					bw.write(1 + "\n");
				}
				else {
					bw.write(0 + "\n");
				}
			}
			else if(str.equals("front")) {
				if(!queue.isEmpty()) {
					bw.write(queue.element() + "\n");
				}
				else {
					bw.write(-1 + "\n");
				}
			}
			else if(str.equals("back")) {
				if(!queue.isEmpty()) {
					bw.write(last + "\n");
				}
				else {
					bw.write(-1 + "\n");
				}
			}
		}
		
		bw.flush();
	}

}

File: 백준/Silver/18258. 큐 2/큐 2.cc
#include <bits/stdc++.h>
using namespace std;

int main(){
    ios_base::sync_with_stdio(false);
    cin.tie(0);

    int N, num;
    string order;
    queue<int> Q;  // 큐 선언

    cin >> N;
    while (N--){
        cin >> order;
        switch (order[1]) {
            case 'u':
                cin >> num;
                Q.push(num);
                break;
            case 'o':
                if (Q.empty()) cout << -1 << '\n';
                else {
                    cout << Q.front() << '\n';
                    Q.pop();
                }
                break;
            case 'i':
                cout<<Q.size()<<'\n';
                break;
            case 'm':
                if (Q.empty()) cout << 1 << '\n';
                else cout << 0 << '\n';
                break;
            case 'r':
                if (Q.empty()) cout << -1 << '\n';
                else cout << Q.front() << '\n';
                break;
            case 'a':
                if (Q.empty()) cout << -1 << '\n';
                else cout << Q.back() << '\n';
                break;
            default:
                break;
        }
    }
    
    return 0;
}